cirq/ionq/job.py

```diff
@@ -154,15 +148,25 @@ def repetitions(self) -> Optional[int]:
         If run on the simulator this will return None.
 
         Raises:
-            IonQUnsuccessfulJob: If the job has failed, been canceled, or
-                deleted.
+            IonQUnsuccessfulJob: If the job has failed, been canceled, or deleted.
             IonQException: If unable to get the status of the job from the API.
         """
         self._check_if_unsuccessful()
         if 'metadata' in self._job and 'shots' in self._job['metadata']:
             return int(self._job['metadata']['shots'])
         return None
 
+    def measurement_dict(self) -> Dict[str, Sequence[int]]:
+        """Returns a dictionary of measurement keys to target qubit index."""
+        measurement_dict: Dict[str, Sequence[int]] = {}
+        if 'metadata' in self._job:
+            for key, value in self._job['metadata'].items():
+                # If we had to pass shots, ignore, it cannot be a measurement key.
+                if key == 'shots':
```

Comment:
what if we use a specific key for measurement keys, instead of them being in the root, so it would look something like this:
`{ "shots": 100, "measurements": { ... } }`  Just to avoid this "special" handling in case we extend metadata with more fields.

Reply:
That's a good idea.  The issue mostly is that really limits measurement key size plus number of qubits and this starts to get very close to cases that people might want to do with a larger qc.  Instead I will make multiple measurementX keys and then just a giant serialized string.

Reply:
I should also mention that doing it this way does add a restriction on the key, since it needs to be in some delimitated form.